core/src/main/java/org/apache/calcite/adapter/enumerable/RexToLixTranslator.java

```diff
@@ -973,6 +973,9 @@ private static Expression scaleIntervalToNumber(
       RelDataType sourceType,
       RelDataType targetType,
       Expression operand) {
+    if (targetType.getSqlTypeName().getFamily() == null) {
+      return operand;
+    }
     switch (requireNonNull(targetType.getSqlTypeName().getFamily(),
         () -> "SqlTypeFamily for " + targetType)) {
```

Comment:
Please rework, and remove code duplication.
Extract `family` into a regular variable, and perform a regular null check. Then `requireNonNull` will not be needed.

Thank you so much for doing so.

Reply:
@vlsi thanks for reviewing. agree with you. I have fixed it. If you have time, please take a look. thanks

Reply:
@vlsi hi, considering [calcite-5960](https://issues.apache.org/jira/browse/CALCITE-5960) has been resolved. Now this PR has nothing to do with calcite-5960. thanks again.